pkg/apiserver/experiment/experiment.go

```diff
@@ -54,187 +88,104 @@ func Register(r *gin.RouterGroup, s *Service) {
-			stateInfo["Finished"]++
-		}
-	}
-	return nil
+	c.JSON(http.StatusOK, data)
 }
 
-// getKernelChaosState returns the state of KernelChaos
-func (s *Service) getKernelChaosState(stateInfo map[string]int) error {
-	var chaosList v1alpha1.KernelChaosList
-	err := s.kubeCli.List(context.Background(), &chaosList)
-	if err != nil {
-		return err
-	}
-	for _, chaos := range chaosList.Items {
-		stateInfo[string(chaos.Status.ChaosStatus.Experiment.Phase)]++
-	}
-	stateInfo["Total"] += len(chaosList.Items)
-
-	return nil
-}
+// TODO: need to be implemented
+func (s *Service) getExperimentDetail(c *gin.Context) {}
 
-// getStressChaosState returns the state of StressChaos
-func (s *Service) getStressChaosState(stateInfo map[string]int) error {
-	var chaosList v1alpha1.StressChaosList
-	err := s.kubeCli.List(context.Background(), &chaosList)
-	if err != nil {
-		return err
-	}
-	for _, chaos := range chaosList.Items {
-		stateInfo["Total"]++
-		switch chaos.Status.ChaosStatus.Experiment.Phase {
-		case v1alpha1.ExperimentPhaseRunning:
-			stateInfo["Running"]++
-		case v1alpha1.ExperimentPhasePaused:
-			stateInfo["Paused"]++
-		case v1alpha1.ExperimentPhaseFailed:
-			stateInfo["Failed"]++
-		case v1alpha1.ExperimentPhaseFinished:
-			stateInfo["Finished"]++
-		}
-	}
-	return nil
-}
+// TODO: need to be implemented
+func (s *Service) deleteExperiment(c *gin.Context) {}
 
+// @Summary Get chaos experiments state from Kubernetes cluster.
+// @Description Get chaos experiments state from Kubernetes cluster.
+// @Tags experiments
+// @Produce json
+// @Success 200 {object} ChaosState
+// @Router /api/experiments/state [get]
+// @Failure 500 {object} utils.APIError
 func (s *Service) state(c *gin.Context) {
-	data := make(map[string]int)
-	data["Total"] = 0
-	data["Running"] = 0
-	data["Paused"] = 0
-	data["Failed"] = 0
-	data["Finished"] = 0
-
-	err := s.getPodChaosState(data)
-	if err != nil {
-		_ = c.Error(err)
-		return
-	}
-	err = s.getIoChaosState(data)
-	if err != nil {
-		_ = c.Error(err)
-		return
-	}
-	err = s.getNetworkChaosState(data)
-	if err != nil {
-		_ = c.Error(err)
-		return
-	}
-	err = s.getTimeChaosState(data)
-	if err != nil {
-		_ = c.Error(err)
-		return
-	}
-	err = s.getKernelChaosState(data)
-	if err != nil {
-		_ = c.Error(err)
-		return
-	}
-	err = s.getStressChaosState(data)
-	if err != nil {
-		_ = c.Error(err)
+	data := new(ChaosState)
+
+	g, ctx := errgroup.WithContext(context.Background())
+	m := &sync.Mutex{}
+	for index := range kinds {
+		list := kinds[index]
+		g.Go(func() error {
+			if err := s.kubeCli.List(ctx, list); err != nil {
+				return err
+			}
+			m.Lock()
+			for _, chaos := range list.ListChaos() {
+				switch chaos.Status {
+				case string(v1alpha1.ExperimentPhaseRunning):
+					data.Running++
+				case string(v1alpha1.ExperimentPhasePaused):
+					data.Paused++
+				case string(v1alpha1.ExperimentPhaseFailed):
+					data.Failed++
+				case string(v1alpha1.ExperimentPhaseFinished):
+					data.Finished++
```

Comment:
Seems swaggo doesn't support to return a map? So I just use a struct to keep the number of each state. 

```
@Success 200 {object} ChaosState
```